src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -85,9 +85,23 @@ export function useSetupTradeState(): void {
    */
   useEffect(() => {
     if (isChainIdSet || !chainIdFromUrl || !currentChainId) return
+    let isSubscribed = true
 
     setIsChainIdSet(true)
-    switchChain(connector, chainIdFromUrl).finally(updateStateAndNavigate)
+
+    switchChain(connector, chainIdFromUrl)
+      .catch((e) => {
+        console.error(e)
+      })
+      .finally(() => {
+        if (!isSubscribed) return
```

Comment:
If I understand correctly, this is so that if `switchChain` finishes when the hook is already unmounted you don't want to update the state and navigate?

Reply:
Yes, you are right